source/scratch/os.hpp

```diff
@@ -8,6 +8,172 @@
 #include <ogc/lwp_watchdog.h>
 #include <ogc/system.h>
 #endif
+#ifdef __PS3__
+#include <sys/systime.h>
+#endif
+
+class MemoryTracker {
+  private:
+    static size_t totalAllocated;
+    static size_t totalVRAMAllocated;
+    static size_t peakUsage;
+    static size_t allocationCount;
+
+    // ---- Max RAM values (max usable ram minus a slight safety margin) ----
+    const static size_t old3ds_maxRamUsage = 50331648;   // 48 MB
+    const static size_t new3ds_maxRamUsage = 100663296;  // 96 MB
+    const static size_t wiiu_maxRamUsage = 805306368;    // 768 MB
+    const static size_t wii_maxRamUsage = 86900736;      // 83 MB
+    const static size_t gamecube_maxRamUsage = 23068672; // 22 MB
+    const static size_t pc_maxRamUsage = 1073741824;     // 1 GB
+    const static size_t vita_maxRamUsage = 335544320;    // 320 MB
+    const static size_t ps4_maxRamUsage = 1073741824;    // 1 GB
+    const static size_t ps3_maxRamUsage = (254*1024*1024)-2;    // 255 MB
```

Comment:
Maybe make this 192 MB instead? Other platforms don't use the exact RAM count. (The Vita for example is 320 MB despite having 512 MB of RAM)